### PR TITLE
BK-3603: [CLI] All commands return items/lists

### PR DESCRIPTION
**Pull request by eyal-stratoscale**
Make all commands return items or lists that can be
represented in a table.
- Make all commands accept the -f flag.
- Return the value 'Success' in case that the
  returned value is None or an empty string.

@Stratoscale/mgmt

### CODE REVIEW

File: hammock/converters.py [hunk 1]
```diff
@@ -45,5 +45,5 @@ def to_str(value):
     return str(value) if value is not None else None
 
 
-def to_none(value):  # pylint: disable=unused-argument
+def to_none(_):
```
Comment (shoham-stratoscale) [inline hammock/converters.py]: lol i didn't know this was legal

File: hammock/cli/command.py [hunk 2]
```diff
@@ -121,6 +119,12 @@ class CommandItem(Command, show.ShowOne):
 
     def take_action(self, parsed_args):
         result = munch.unmunchify(self._action(parsed_args))
+        if result is not None and (self.spec.returns is not None):
+            result = self.spec.returns.convert(result)
+        if result is None or result == '':
+            result = 'Success'
+        if not isinstance(result, dict):
```
Comment (shoham-stratoscale) [inline hammock/cli/command.py]: What if convert() will return a non-dict? Do you want it converted?

File: hammock/cli/command.py [hunk 3]
```diff
@@ -27,13 +27,11 @@ def factory(func, column_order=None, column_colors=None):
     }
 
     if spec.returns:
-        if spec.returns.type_name == 'dict':
```
Comment (shoham-stratoscale) [inline hammock/cli/command.py]: why remove dict?

File: hammock/cli/command.py [hunk 4]
```diff
@@ -27,13 +27,11 @@ def factory(func, column_order=None, column_colors=None):
...
-            return type(func.__name__, (CommandItem, ), overrides)
-        elif spec.returns.type_name == 'list':
+        if spec.returns.type_name == 'list':
             return type(func.__name__, (CommandList, ), overrides)
         elif spec.returns.type_name == 'file':
             return type(func.__name__, (CommandFile, ), overrides)
-    return type(func.__name__, (Command, ), overrides)
```
Comment (shoham-stratoscale) [inline hammock/cli/command.py]: Will it always be now a "single item"?

Comment (shoham-stratoscale) [inline hammock/cli/command.py]: what about "if dict -> return CommandItem, else -> Exception"?

**Comment by shoham-stratoscale** [comment]
shipit

File: hammock/cli/command.py (hunk 4 above)
Comment (eyal-stratoscale) [inline hammock/cli/command.py]: but i want to return it also on other types...

Comment (lior-stratoscale) [inline hammock/cli/command.py]: why didn't you use a dictionary here?

Comment (eyal-stratoscale) [inline hammock/cli/command.py]: what do you mean?

**Pull request merged**